sys/errors.h

```diff
@@ -8,6 +8,7 @@ typedef enum {
     ERR_CODE_UNKNOWN = 1,
     ERR_CODE_INVALID_ARG = 2,
     ERR_CODE_INVALID_STATE = 3,
+    NULL_ERR= 3,
```

Comment:
nit: same comment as above about naming consistency

Reply:
Also the value should be 4 instead of 3 to avoid overlap between this error code and invalid state